ansible_rulebook/engine.py

```diff
@@ -318,74 +206,278 @@ async def run_rulesets(
         task.cancel()
 
 
-async def run_ruleset(
-    event_log: asyncio.Queue,
-    ruleset_queue_plan: EngineRuleSetQueuePlan,
-    hosts_facts: List[Dict],
-    variables: Dict,
-    project_data_file: Optional[str] = None,
-):
-
-    name = ruleset_queue_plan.ruleset.name
+class RuleSetRunner:
+    def __init__(
+        self,
+        event_log: asyncio.Queue,
+        ruleset_queue_plan: EngineRuleSetQueuePlan,
+        hosts_facts,
+        variables,
+        project_data_file: Optional[str] = None,
+    ):
+        self.pa_runner = PlaybookActionRunner()
+        self.pa_runner_task = None
+        self.action_tasks = []
+        self.event_log = event_log
+        self.ruleset_queue_plan = ruleset_queue_plan
+        self.name = ruleset_queue_plan.ruleset.name
+        self.hosts_facts = hosts_facts
+        self.variables = variables
+        self.project_data_file = project_data_file
+
+    async def run_ruleset(self):
+        prime_facts(self.name, self.hosts_facts, self.variables)
+        self.pa_runner_task = asyncio.create_task(
```

Comment:
So far we still cannot sure the action in the current ruleset has `run_playbook` or not, right? is it too earlier to create a playbook runner task? should we wait to do this until we know we have `run_playbook` action, like the calling of `wait_for_playbook`?

Reply:
We could do a lazy initialization, but since the main use of ansible-rulebook is to run playbooks I feel the code is simple and easy to read by creating the runner task up the front.